File: nand2tetris/projects/06/Assembler/src/Assembler.java
public class Assembler
{
    static int variablePlacement;
    private Parser parser;

    Assembler(String fileName) // filename will have it in the name NAME.asm
    {
        int findDotASM = fileName.indexOf(".asm");
        String asmFileName = fileName.substring(0,findDotASM);
        parser = new Parser(asmFileName);
    }
};

File: nand2tetris/projects/06/Nand2TetrisAssembler/Assembler.cpp
//
// Created by Bryce Callender on 2019-03-13.
//

#include "Assembler.h"

int Assembler::variablePlacement = 16;

Assembler::Assembler(std::string &fileName)
{
    unsigned long findDotASM = fileName.find(".asm");
    std::string asmFileName = fileName.substr(0,findDotASM);
    fileParser = new Parser(asmFileName);
}

Assembler::~Assembler()
{
    delete fileParser;
}

